runtime/include/tt/runtime/detail/workarounds.h

```diff
@@ -39,13 +40,19 @@ struct Env {
   // to be able to pluck this update index from a runtime tensor.
   bool readUpdateIndexFromDeviceForKVCache;
 
+  // TODO(bug #1658): We're currently use ttnn::to_dtype operation to cast the
+  // data type of a tensor on host. Once we have improved the typecast operation
+  // to handle this, we should remove this workaround.
+  bool typecastOnHost;
+
 private:
   constexpr Env(bool maxpool2dPreshard, bool swapBinaryOperands,
-                bool readUpdateIndexFromDeviceForKVCache)
+                bool readUpdateIndexFromDeviceForKVCache, bool typecastOnHost)
```

Comment:
Maybe name this as toDtypeOnHost or something so that it's explicit that we're using to_dtype when typcasting on host.

Also please add a ttrt command line option to toggle this in `runtime/tools/python/ttrt/common/run.py`, the procedure will be the same as the other workaround flags.

Reply:
Renamed typecastOnHost to toDtypeOnHost. I added an option in run.py; please double-check if I implemented it correctly.